### ISSUE TITLE
Incompatibility with Drag Ruler in PF2E system

### ISSUE THREAD

**Issue by apoapostolov** [Saibot393/perceptive#2]
I tested Drag Ruler and Perceptive in the PF2 5.4.0 system. Drag Ruler stops working. Every movement generates an error. 

```
PeekingScript.js:169 Uncaught TypeError: Cannot read properties of null (reading 'document')
    at ClockwiseSweepPolygon._testWallInclusion (PeekingScript.js:169:75)
    at collisionTest (foundry.js:34011:45)
    at Quadtree.getObjects (foundry.js:30478:55)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at CanvasQuadtree.getObjects (foundry.js:30471:11)
    at ClockwiseSweepPolygon._getWalls (foundry.js:34012:34)
    at ClockwiseSweepPolygon._identifyEdges (foundry.js:48712:24)
    at ClockwiseSweepPolygon._testCollision (foundry.js:49153:10)
    at ClockwiseSweepPolygon.testCollision (foundry.js:34151:17)
    at stepCollidesWithWall (cache.js:223:44)
    at GriddedCache.getInitializedNode (cache.js:109:10)
    at GriddedPathfinder.reset (pathfinder.js:40:26)
    at new GriddedPathfinder (pathfinder.js:28:8)
    at initializePathfinder (main.js:47:10)
    at Object.calculatePath (main.js:52:21)
    at DragRulerRuler.measure (ruler.js:123:39)
    at DragRulerRuler.scheduleMeasurement (foundry_imports.js:153:8)
    at DragRulerRuler.onMouseMove (foundry_imports.js:145:22)
    at _TokenPF2e.onEntityLeftDragMove (main.js:159:37)
    at _TokenPF2e.onEntityLeftDragMoveSnap (main.js:153:23)
    at 🎁Token.prototype._onDragLeftMove#0 (libWrapper-wrapper.js:187:52)
    at MouseInteractionManager.callback (foundry.js:31655:17)
    at #handleDragMove (foundry.js:32002:26)
    at #handleMouseMove (foundry.js:31974:37)
    at Na.notifyListeners (earcut.js:128:75)
    at Na.notifyTarget (earcut.js:128:75)
    at Na.propagate (earcut.js:128:75)
    at Na.dispatchEvent (earcut.js:128:75)
    at Na.mapPointerMove (earcut.js:128:75)
    at Na.mapEvent (earcut.js:128:75)
    at yn.onPointerMove (earcut.js:128:75)
ClockwiseSweepPolygon._testWallInclusion @ PeekingScript.js:169
collisionTest @ foundry.js:34011
getObjects @ foundry.js:30478
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
_getWalls @ foundry.js:34012
_identifyEdges @ foundry.js:48712
_testCollision @ foundry.js:49153
testCollision @ foundry.js:34151
stepCollidesWithWall @ cache.js:223
getInitializedNode @ cache.js:109
reset @ pathfinder.js:40
GriddedPathfinder @ pathfinder.js:28
initializePathfinder @ main.js:47
calculatePath @ main.js:52
measure @ ruler.js:123
scheduleMeasurement @ foundry_imports.js:153
onMouseMove @ foundry_imports.js:145
onEntityLeftDragMove @ main.js:159
onEntityLeftDragMoveSnap @ main.js:153
🎁Token.prototype._onDragLeftMove#0 @ libWrapper-wrapper.js:187
callback @ foundry.js:31655
#handleDragMove @ foundry.js:32002
#handleMouseMove @ foundry.js:31974
notifyListeners @ earcut.js:128
notifyTarget @ earcut.js:128
propagate @ earcut.js:128
dispatchEvent @ earcut.js:128
mapPointerMove @ earcut.js:128
mapEvent @ earcut.js:128
onPointerMove @ earcut.js:128
3PeekingScript.js:169 Uncaught TypeError: Cannot read properties of null (reading 'document')
    at ClockwiseSweepPolygon._testWallInclusion (PeekingScript.js:169:75)
    at collisionTest (foundry.js:34011:45)
    at Quadtree.getObjects (foundry.js:30478:55)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at CanvasQuadtree.getObjects (foundry.js:30471:11)
    at ClockwiseSweepPolygon._getWalls (foundry.js:34012:34)
    at ClockwiseSweepPolygon._identifyEdges (foundry.js:48712:24)
    at ClockwiseSweepPolygon._testCollision (foundry.js:49153:10)
    at ClockwiseSweepPolygon.testCollision (foundry.js:34151:17)
    at stepCollidesWithWall (cache.js:223:44)
    at GriddedCache.getInitializedNode (cache.js:109:10)
    at GriddedPathfinder.reset (pathfinder.js:40:26)
    at new GriddedPathfinder (pathfinder.js:28:8)
    at initializePathfinder (main.js:47:10)
    at Object.calculatePath (main.js:52:21)
    at DragRulerRuler.measure (ruler.js:123:39)
    at DragRulerRuler.scheduleMeasurement (foundry_imports.js:153:8)
    at DragRulerRuler.onMouseMove (foundry_imports.js:145:22)
    at _TokenPF2e.onEntityLeftDragMove (main.js:159:37)
    at _TokenPF2e.onEntityLeftDragMoveSnap (main.js:153:23)
    at 🎁Token.prototype._onDragLeftMove#0 (libWrapper-wrapper.js:187:52)
    at MouseInteractionManager.callback (foundry.js:31655:17)
    at #handleDragMove (foundry.js:32002:26)
    at #handleMouseMove (foundry.js:31974:37)
    at Na.notifyListeners (earcut.js:128:75)
    at Na.notifyTarget (earcut.js:128:75)
    at Na.propagate (earcut.js:128:75)
    at Na.dispatchEvent (earcut.js:128:75)
    at Na.mapPointerMove (earcut.js:128:75)
    at Na.mapEvent (earcut.js:128:75)
    at yn.onPointerMove (earcut.js:128:75)
ClockwiseSweepPolygon._testWallInclusion @ PeekingScript.js:169
collisionTest @ foundry.js:34011
getObjects @ foundry.js:30478
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
_getWalls @ foundry.js:34012
_identifyEdges @ foundry.js:48712
_testCollision @ foundry.js:49153
testCollision @ foundry.js:34151
stepCollidesWithWall @ cache.js:223
getInitializedNode @ cache.js:109
reset @ pathfinder.js:40
GriddedPathfinder @ pathfinder.js:28
initializePathfinder @ main.js:47
calculatePath @ main.js:52
measure @ ruler.js:123
scheduleMeasurement @ foundry_imports.js:153
onMouseMove @ foundry_imports.js:145
onEntityLeftDragMove @ main.js:159
onEntityLeftDragMoveSnap @ main.js:153
🎁Token.prototype._onDragLeftMove#0 @ libWrapper-wrapper.js:187
callback @ foundry.js:31655
#handleDragMove @ foundry.js:32002
#handleMouseMove @ foundry.js:31974
notifyListeners @ earcut.js:128
notifyTarget @ earcut.js:128
propagate @ earcut.js:128
dispatchEvent @ earcut.js:128
mapPointerMove @ earcut.js:128
mapEvent @ earcut.js:128
onPointerMove @ earcut.js:128
2PeekingScript.js:169 Uncaught TypeError: Cannot read properties of null (reading 'document')
    at ClockwiseSweepPolygon._testWallInclusion (PeekingScript.js:169:75)
    at collisionTest (foundry.js:34011:45)
    at Quadtree.getObjects (foundry.js:30478:55)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at CanvasQuadtree.getObjects (foundry.js:30471:11)
    at ClockwiseSweepPolygon._getWalls (foundry.js:34012:34)
    at ClockwiseSweepPolygon._identifyEdges (foundry.js:48712:24)
    at ClockwiseSweepPolygon._testCollision (foundry.js:49153:10)
    at ClockwiseSweepPolygon.testCollision (foundry.js:34151:17)
    at stepCollidesWithWall (cache.js:223:44)
    at GriddedCache.getInitializedNode (cache.js:109:10)
    at GriddedPathfinder.reset (pathfinder.js:40:26)
    at new GriddedPathfinder (pathfinder.js:28:8)
    at initializePathfinder (main.js:47:10)
    at Object.calculatePath (main.js:52:21)
    at DragRulerRuler.measure (ruler.js:123:39)
    at DragRulerRuler.scheduleMeasurement (foundry_imports.js:153:8)
    at DragRulerRuler.onMouseMove (foundry_imports.js:145:22)
    at _TokenPF2e.onEntityLeftDragMove (main.js:159:37)
    at _TokenPF2e.onEntityLeftDragMoveSnap (main.js:153:23)
    at 🎁Token.prototype._onDragLeftMove#0 (libWrapper-wrapper.js:187:52)
    at MouseInteractionManager.callback (foundry.js:31655:17)
    at #handleDragMove (foundry.js:32002:26)
    at #handleMouseMove (foundry.js:31974:37)
    at Na.notifyListeners (earcut.js:128:75)
    at Na.notifyTarget (earcut.js:128:75)
    at Na.propagate (earcut.js:128:75)
    at Na.dispatchEvent (earcut.js:128:75)
    at Na.mapPointerMove (earcut.js:128:75)
    at Na.mapEvent (earcut.js:128:75)
    at yn.onPointerMove (earcut.js:128:75)
ClockwiseSweepPolygon._testWallInclusion @ PeekingScript.js:169
collisionTest @ foundry.js:34011
getObjects @ foundry.js:30478
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
_getWalls @ foundry.js:34012
_identifyEdges @ foundry.js:48712
_testCollision @ foundry.js:49153
testCollision @ foundry.js:34151
stepCollidesWithWall @ cache.js:223
getInitializedNode @ cache.js:109
reset @ pathfinder.js:40
GriddedPathfinder @ pathfinder.js:28
initializePathfinder @ main.js:47
calculatePath @ main.js:52
measure @ ruler.js:123
scheduleMeasurement @ foundry_imports.js:153
onMouseMove @ foundry_imports.js:145
onEntityLeftDragMove @ main.js:159
onEntityLeftDragMoveSnap @ main.js:153
🎁Token.prototype._onDragLeftMove#0 @ libWrapper-wrapper.js:187
callback @ foundry.js:31655
#handleDragMove @ foundry.js:32002
#handleMouseMove @ foundry.js:31974
notifyListeners @ earcut.js:128
notifyTarget @ earcut.js:128
propagate @ earcut.js:128
dispatchEvent @ earcut.js:128
mapPointerMove @ earcut.js:128
mapEvent @ earcut.js:128
onPointerMove @ earcut.js:128
2PeekingScript.js:169 Uncaught TypeError: Cannot read properties of null (reading 'document')
    at ClockwiseSweepPolygon._testWallInclusion (PeekingScript.js:169:75)
    at collisionTest (foundry.js:34011:45)
    at Quadtree.getObjects (foundry.js:30478:55)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at Quadtree.getObjects (foundry.js:30471:11)
    at CanvasQuadtree.getObjects (foundry.js:30471:11)
    at ClockwiseSweepPolygon._getWalls (foundry.js:34012:34)
    at ClockwiseSweepPolygon._identifyEdges (foundry.js:48712:24)
    at ClockwiseSweepPolygon._testCollision (foundry.js:49153:10)
    at ClockwiseSweepPolygon.testCollision (foundry.js:34151:17)
    at stepCollidesWithWall (cache.js:223:44)
    at GriddedCache.getInitializedNode (cache.js:109:10)
    at GriddedPathfinder.reset (pathfinder.js:40:26)
    at new GriddedPathfinder (pathfinder.js:28:8)
    at initializePathfinder (main.js:47:10)
    at Object.calculatePath (main.js:52:21)
    at DragRulerRuler.measure (ruler.js:123:39)
    at DragRulerRuler.scheduleMeasurement (foundry_imports.js:153:8)
    at DragRulerRuler.onMouseMove (foundry_imports.js:145:22)
    at _TokenPF2e.onEntityLeftDragMove (main.js:159:37)
    at _TokenPF2e.onEntityLeftDragMoveSnap (main.js:153:23)
    at 🎁Token.prototype._onDragLeftMove#0 (libWrapper-wrapper.js:187:52)
    at MouseInteractionManager.callback (foundry.js:31655:17)
    at #handleDragMove (foundry.js:32002:26)
    at #handleMouseMove (foundry.js:31974:37)
    at Na.notifyListeners (earcut.js:128:75)
    at Na.notifyTarget (earcut.js:128:75)
    at Na.propagate (earcut.js:128:75)
    at Na.dispatchEvent (earcut.js:128:75)
    at Na.mapPointerMove (earcut.js:128:75)
    at Na.mapEvent (earcut.js:128:75)
    at yn.onPointerMove (earcut.js:128:75)
ClockwiseSweepPolygon._testWallInclusion @ PeekingScript.js:169
collisionTest @ foundry.js:34011
getObjects @ foundry.js:30478
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
getObjects @ foundry.js:30471
_getWalls @ foundry.js:34012
_identifyEdges @ foundry.js:48712
_testCollision @ foundry.js:49153
testCollision @ foundry.js:34151
stepCollidesWithWall @ cache.js:223
getInitializedNode @ cache.js:109
reset @ pathfinder.js:40
GriddedPathfinder @ pathfinder.js:28
initializePathfinder @ main.js:47
calculatePath @ main.js:52
measure @ ruler.js:123
scheduleMeasurement @ foundry_imports.js:153
onMouseMove @ foundry_imports.js:145
onEntityLeftDragMove @ main.js:159
onEntityLeftDragMoveSnap @ main.js:153
🎁Token.prototype._onDragLeftMove#0 @ libWrapper-wrapper.js:187
callback @ foundry.js:31655
#handleDragMove @ foundry.js:32002
#handleMouseMove @ foundry.js:31974
notifyListeners @ earcut.js:128
notifyTarget @ earcut.js:128
propagate @ earcut.js:128
dispatchEvent @ earcut.js:128
mapPointerMove @ earcut.js:128
mapEvent @ earcut.js:128
onPointerMove @ earcut.js:128
game:1 Unchecked runtime.lastError: The message port closed before a response was received.
```

**Comment by Saibot393** [comment]
Thanks for the report, i will look into it.

**Comment by Saibot393** [comment]
I can not replicate this bug. I will however still be able to implement a preliminary fix (thanks for including the error).
Some questions to narrow down possible causes:
1. Which Drag ruler version are you using?
2. Which Foundry version are you using?
3. What other modules were active when the error occurred?
4. Were you logged in as a play or as a GM when the error occurred?

**Comment by apoapostolov** [comment]
Drag Ruler 1.13.8 , Foundry Build 309 latest, I was logged as a GM

The mod list is below but I resplicated it with no mods via culrpit

### Active Modules
Accessibility Enhancements
Actually Private Messages (ex Actually Private Rolls)
Add Party to Combat
Advanced Macros
Ambient Soundbits
Automatic Journal Icon Numbers
Bar Brawl
Border Control
Boss Bar
Cautious Gamemaster's Pack
🧼 Chat Cleaner
Chat GIFs
Combat Initiative Modifier
Combat Music Master
ConversationHUD
Custom CSS
Damage Log
Death Mark
DFreds Droppables
Dice So Nice!
Dice Stats
Dice Tray
Die Hard
Drag Ruler
PF2e Drag Ruler Integration
Dramatic Rolls
Dynamic Soundscapes
Easy Notes
Find the culprit
FXMaster
GM Screen
Heartbeat
Hidden Soundtracks
Hide GM Rolls
Initiative Double Click
Item Piles
JB2A - Patreon Complete Collection
CodeMirror
lib - Color Settings
Forien's Copy Environment
libWrapper
socketlib
Token Action HUD Core
Lock View
Module Management+
Monk's Combat Details
Monk's Combat Marker
Monk's Hotbar Expansion
Monk's Little Details
Monk's Player Settings
Monk's TokenBar
Monk's Wall Enhancement
Moulinette Core
Moulinette Tiles (module)
Narrator Tools
Party Overview
Pathfinder 2e: Abomination Vaults
Pathfinder Token Pack: Bestiaries
PF2e Action Support
PF2e Alignment Damage and Other Immunities
PF2e Companion Compendia
PF2e Convenient Consumables
PF2e Dailies
PF2E Exploration Effects
pf2E Extempore Effects
PF2e Flat Check
PF2e Giveth
PF2e Interactive Token Tooltip
PF2e Keybind Menagerie
Pf2E Modifiers Matter
PF2e Party Sheet Helper
PF2e Perception
PF2e Reaction Checker
PF2e Rogue's Initiative
PF2e Rule Element Generator
PF2e Spells Summary
PF2e Target Damage
pf2e Utility Buttons
PF2e Workbench
PF2 Shields Token Bar
Pin Cushion
Playlist Enhancer
PnP - Pointer and Pings!
PopOut!
Quick Insert - Search Widget
Rounded Distance for Measured Templates
routinglib
Scrolling Texts
Sequencer
Automated Animations
PF2e Animation Macros
Share Media
Simple Calendar
SmallTime
Small Visual Tweaks
Smart Doors
Smart Target
SoundFx Library
Splatter
Status Icon Tweaks
Target Improvements
Token Action HUD Pathfinder 2e
Token Auras
Token Ease
Token Health
Tokenizer
Token Lean: Peek Around Corners
Token Light Condition
Token Magic FX
PF2e TokenMagic Integration
Token Z
Torch
True RNG
Tsamys Secret Rolls
PF2e Dorako UI
Warp Gate
Foundry Summons
Z Scatter

**Comment by Saibot393** [comment]
I just released v1.3.0. I think that should fix the bug, though i can still not replicate the bug and have therefore not been able to test it.

**Comment by Saibot393** [comment]
I was finally able to replicate the bug, it only seems to appear on gridded maps. It should be fixed with v1.2.2

**Comment by apoapostolov** [comment]
It is fixed, thank you!
